File: src/main/webapp/static/js/service/pixelmapping/refer-url-key-mappings.service.js
/**
 * @author YI ZHAO[yi.zhao@adara.com]
 */

'use strict';

app.factory("referUrlKeyMappingsService", ['$http', '$location', '$rootScope',
    function ($http, $location, $rootScope) {
        var service = {};

        // data mapping services
        service.getMappings = getMappings;
        service.getMapping = getMapping;
        service.insertMapping = insertMapping;
        service.updateMapping = updateMapping;
        service.deleteMapping = deleteMapping;

        /**
         *   data mapping services
         **/
        function getMappings() {
            var body = $http.get($rootScope.base + 'mappings/rukm');
            return body;
        }

        function getMapping(mappingID) {
            return $http.get($rootScope.base + 'mapping/rukm?id=' + mappingID);
        }

        function insertMapping(redirectPath, frontendData) {
            return $http.post($rootScope.base + 'insertMapping/rukm', {dp_id: frontendData.dp_id, refer_url_key_id: frontendData.refer_url_key_id, enabled: frontendData.enabled}).success(function (results) {
                $location.path(redirectPath);
                return results;
            });
        };

        function updateMapping(redirectPath, id, frontendData) {
            return $http.post($rootScope.base + 'updateMapping/rukm', {dp_id: frontendData.dp_id, refer_url_key_id: frontendData.refer_url_key_id, enabled: frontendData.enabled}).success(function (status) {
                $location.path(redirectPath);
                return status.data;
            });
        };

        function deleteMapping(redirectPath, id) {
            return $http.delete($rootScope.base + 'deleteMapping/rukm?id=' + id).success(function (status) {
                $location.path(redirectPath);
                return status.data;
            });
        };

        return service;
    }
]);

File: src/main/java/com/adara/pixeldataengineui/service/pixelmapping/ReferUrlKeyMappingsService.java
package com.adara.pixeldataengineui.service.pixelmapping;

import com.adara.pixeldataengineui.model.backend.dto.generic.GenericDTOList;
import com.adara.pixeldataengineui.model.backend.dto.generic.ResponseDTO;
import com.adara.pixeldataengineui.model.backend.dto.pixelmapping.ReferUrlKeyMappingsDTO;

/**
 * Created by yzhao on 10/12/16.
 */
public interface ReferUrlKeyMappingsService {
    GenericDTOList<ReferUrlKeyMappingsDTO> getMappings() throws Exception;

    ReferUrlKeyMappingsDTO getMapping(String dpId) throws Exception;

    ResponseDTO insertMapping(ReferUrlKeyMappingsDTO request) throws Exception;

    ResponseDTO updateMapping(ReferUrlKeyMappingsDTO request) throws Exception;

    ResponseDTO deleteMapping(String dpId) throws Exception;
}
